openmmtools/multistate/multistatereporter.py

```diff
@@ -368,7 +368,7 @@ def _open_dataset_robustly(self, *args, n_attempts=5, sleep_time=2,
         for attempt in range(n_attempts-1):
             try:
                 return netcdf.Dataset(*args, **kwargs)
-            except:
+            except: # This should be safe since we will raise the error below on return
```

Comment:
I think we still want to output the error/exception message here in the debug log, since it could be potentially different than the one being generated on return. Does that make sense?

Reply:
Sure! We can log it